src/Component/tests/Doctrine/Persistence/InMemoryRepositoryTest.php

```diff
@@ -0,0 +1,266 @@
+        $this->repository->add($wrongIdBook->reveal());
+        $this->repository->add($wrongNameBook->reveal());
+
+        $this->assertSame(
+            [$thirdBook->reveal(), $firstBook->reveal()],
+            $this->repository->findBy(
+                ['name' => $name, 'id' => $id],
+                ['rating' => RepositoryInterface::ORDER_ASCENDING, 'title' => RepositoryInterface::ORDER_DESCENDING],
+                2,
+                1,
+            ),
+        );
+    }
+
+    public function testItThrowsInvalidArgumentExceptionWhenFindingOneObjectWithEmptyParameterArray(): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->repository->findOneBy([]);
+    }
+
+    public function testItFindsOneObjectByParameter(): void
+    {
+        $book = $this->prophesize(SampleBookResourceInterface::class);
+        $shirt = $this->prophesize(SampleBookResourceInterface::class);
+
+        $book->getName()->willReturn('Book');
+        $shirt->getName()->willReturn('Shirt');
+
+        $this->repository->add($book->reveal());
+        $this->repository->add($shirt->reveal());
+
+        $this->assertSame($book->reveal(), $this->repository->findOneBy(['name' => 'Book']));
+    }
+
+    public function testItReturnsFirstResultWhileFindingOneByParameters(): void
+    {
+        $book = $this->prophesize(SampleBookResourceInterface::class);
+        $secondBook = $this->prophesize(SampleBookResourceInterface::class);
+
+        $book->getName()->willReturn('Book');
+        $secondBook->getName()->willReturn('Book');
+
+        $this->repository->add($book->reveal());
+        $this->repository->add($secondBook->reveal());
+
+        $this->assertSame($book->reveal(), $this->repository->findOneBy(['name' => 'Book']));
+    }
+
+    public function testItFindsAllObjectsInMemory(): void
+    {
+        $book = $this->prophesize(SampleBookResourceInterface::class);
+        $shirt = $this->prophesize(SampleBookResourceInterface::class);
+
+        $this->repository->add($book->reveal());
+        $this->repository->add($shirt->reveal());
+
+        $this->assertSame([$book->reveal(), $shirt->reveal()], $this->repository->findAll());
+    }
+
+    public function testItReturnsEmptyArrayWhenMemoryIsEmpty(): void
+    {
+        $this->assertSame([], $this->repository->findAll());
+    }
+
+    public function testItCreatesPaginator(): void
+    {
+        $this->assertInstanceOf(Pagerfanta::class, $this->repository->createPaginator());
+    }
+
+    public function testItReturnsStatedClassName(): void
+    {
+        $this->assertSame(SampleBookResourceInterface::class, $this->repository->getClassName());
+    }
+}
+
+interface SampleBookResourceInterface extends ResourceInterface
+{
+    /**
+     * @return string
+     */
+    public function getName();
+
+    /**
+     * @return int
+     */
+    public function getRating();
+
+    /**
+     * @return string
+     */
+    public function getTitle();
```

Comment:
Does it work for tests to make this change?
```suggestion
    public function getName(): string;

    public function getRating(): int;

    public function getTitle(): string;
```